tests/test_brown.py

```diff
@@ -26,48 +27,155 @@
-    jsonrecord["uri"] = "brown/test_splash.html"
 
+    def patch_get_mime_type(*args, **kwargs):
+        """Mock the mime type getting."""
+        return "patched application/pdf"
+
+    monkeypatch.setattr(
+        hepcrawl.spiders.brown_spider,
+        'get_mime_type',
+        patch_get_mime_type
+    )
+    request = spider.parse(
+        fake_response_from_file('brown/test_1.json')
+    ).next()
+    response = HtmlResponse(
+        url=request.url,
+        request=request,
+        body=brown_splash_body,
+        encoding='utf-8',
+    )
+
+    record = request.callback(response)
+
+    # Return the final record after it has gone through the pipeline
+    pipeline = InspireAPIPushPipeline()
+    final_record = pipeline.process_item(record, spider)
 
-    splash_response = fake_response_from_file('brown/test_splash.html')
-    splash_response.meta["jsonrecord"] = jsonrecord
-    parsed_record = spider.scrape_splash(splash_response)
-    assert parsed_record
-    return parsed_record
+    return final_record
 
 
 @pytest.fixture
-def parsed_node():
-    """Return a parse call to a full record.
+def record_no_year_no_author(no_year_no_author_splash, monkeypatch):
+    """Return results from the Brown spider."""
+    spider = brown_spider.BrownSpider()
 
-    Return type should be a Scrapy Request object.
-    """
+    def patch_get_mime_type(*args, **kwargs):
+        """Mock the mime type getting."""
+        return "patched application/pdf"
+
+    monkeypatch.setattr(
+        hepcrawl.spiders.brown_spider,
+        'get_mime_type',
+        patch_get_mime_type
+    )
+    request = spider.parse(
+        fake_response_from_file('brown/test_1.json')
+    ).next()
+    response = HtmlResponse(
+        url=request.url,
+        request=request,
+        body=no_year_no_author_splash,
+        encoding='utf-8',
+    )
+
+    record = request.callback(response)
+    pipeline = InspireAPIPushPipeline()
+    final_record = pipeline.process_item(record, spider)
+
+    return final_record
+
+
+@pytest.fixture
+def record_without_splash(brown_splash_body):
+    """Return a record without splash page url."""
     spider = brown_spider.BrownSpider()
-    response = fake_response_from_file('brown/test_1.json')
+    body = """
+    {
+    "items": {
+        "docs": [
+            {
+                "json_uri": "https://repository.library.brown.edu/api/pub/items/bdr:11303/"
+            }
+        ]
+    }
+    }
+    """
+
+    response = fake_response_from_string(body)
     jsonresponse = json.loads(response.body_as_unicode())
     jsonrecord = jsonresponse["items"]["docs"][0]
     response.meta["jsonrecord"] = jsonrecord
 
-    return spider.parse(response).next()
+    record = spider.parse(response).next()
+    pipeline = InspireAPIPushPipeline()
+    final_record = pipeline.process_item(record, spider)
 
-def test_files_constructed(parsed_node):
-    """Test pdf link.
+    return final_record
```

Comment:
Maybe we can refactor this to be reused on the tests, as it's repeated everywhere. Give it a couple hours, if you don't find a way to improve it, let's merge this and go for the arXiv one.